src/ferui-components/tree-view/tree-view-component.ts

```diff
@@ -0,0 +1,382 @@
+        return childResult;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Handles the expansion of a Tree Node component
+   * @param {TreeNode<T>} node
+   * @param {number} firstIdxInView
+   * @param {number} lastIdxInView
+   * @returns {Promise<void>}
+   */
+  private async handleExpandNode(node: TreeNode<T>, firstIdxInView: number, lastIdxInView: number): Promise<void> {
+    node.showLoader = node.expanded = true;
+    await this.loadMoreNodes(node, this.bufferAmount + lastIdxInView - firstIdxInView, false);
+    node.showLoader = false;
+    this.rebuildVirtualScrollerArray();
+  }
+
+  /**
+   * Handles the collapse event of a Tree Node Component
+   * @param {TreeNode<T>} node
+   * @returns {Promise<void>}
+   */
+  private async handleCollapseNode(node: TreeNode<T>): Promise<void> {
+    node.children = [];
+    node.expanded = node.showLoader = node.loadError = false;
+    node.allChildrenLoaded = false;
+    this.rebuildVirtualScrollerArray();
+  }
+
+  /**
+   * Handles the scroll event of the Tree View
+   * @param {number} lastIdxInView
+   * @returns {Promise<void>}
+   */
+  private async handleScroll(lastIdxInView: number): Promise<void> {
+    const numberNeeded = this.bufferAmount - (this.scrollViewArray.length - lastIdxInView);
+    if (numberNeeded > 0 && !this.scrollPromise) {
+      this.scrollPromise = true;
+      await this.loadMoreNodes(this.scrollViewArray[lastIdxInView].parent, numberNeeded, true);
+      this.rebuildVirtualScrollerArray();
+    }
+    this.cd.markForCheck();
+  }
+
+  /**
+   * Load more Tree Nodes
+   *
+   * @param {TreeNode<T>} node
+   * @param {number} numberToLoad the number of new nodes wanted
+   * @param {boolean} recurse when true, continues to load up to the number of nodes needed from the next
+   * expanded node that has more children or the full set of tree nodes is loaded from current node. When false, only
+   * attempts to load up to 'numberToLoad' from the selected node with an incomplete set of children.
+   * @returns {Promise<void>}
+   */
+  private async loadMoreNodes(node: TreeNode<T>, numberToLoad: number, recurse: boolean) {
+    const firstNodeWithMoreChildrenToLoad = this.getFirstNodeWithMoreChildrenToLoad(node);
+    if (firstNodeWithMoreChildrenToLoad === null) {
+      return;
+    }
+    let newChildren;
+    if (this.serverSideComponent) {
+      try {
+        newChildren = (await (this.dataRetriever as PagedTreeNodeDataRetriever<T>).getPagedChildNodeData(
+          firstNodeWithMoreChildrenToLoad.data,
+          {
+            offset: firstNodeWithMoreChildrenToLoad.children.length,
+            limit: numberToLoad,
+          }
+        )).map(it => this.createTreeNode(it, firstNodeWithMoreChildrenToLoad));
+      } catch (e) {
+        firstNodeWithMoreChildrenToLoad.showLoader = false;
+        firstNodeWithMoreChildrenToLoad.loadError = firstNodeWithMoreChildrenToLoad.allChildrenLoaded = true;
+        this.cd.markForCheck();
```

Comment:
This should be called only if you assign a new value to a variable. You should actually move this to line 339.